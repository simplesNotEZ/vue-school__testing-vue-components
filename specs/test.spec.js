import { mount, shallowMount } from '@vue/test-utils';
import TestComponent from '@/test';
import List from '@/list';

test('mount the TestComponent', () => {
  const wrapper = mount(TestComponent, {
    propsData: {
      value: 'FlueSchool'
    }
  });
  console.log('wrapper.attributes(): ', wrapper.attributes());
  console.log('wrapper.html(): ', wrapper.html());
  console.log(
    "wrapper.find('h1').attributes(): ",
    wrapper.find('h1').attributes()
  );
  console.log('wrapper.vm.$props.value: ', wrapper.vm.$props.value);
  console.log('wrapper.vm.value: ', wrapper.vm.value);
  console.log('wrapper.vm.$data: ', wrapper.vm.$data);
  console.log('wrapper.vm.$data.marvelMovies: ', wrapper.vm.$data.marvelMovies);
  console.log('wrapper.vm.marvelMovies: ', wrapper.vm.marvelMovies);

  // console.log('wrapper.vm........', wrapper.vm);

  expect(wrapper.html()).toMatchSnapshot();
});

test('shallowMount vs mount of List component', () => {
  console.log('mount: ', mount(List).html());
  console.log('shallowMount: ', shallowMount(List).html());
});

test('ListComponent', async () => {
  const wrapper = mount(List);
  const movies = wrapper.vm.marvelMovies;
  wrapper.setData({ marvelMovies: [...movies, 'Resevoir Doggies'] });
  await wrapper.vm.$nextTick();
  // we can simplify the below, because we have installed the jest-serializer-vue package,
  //   which "prettifies" the snapshot for us
  // expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
  expect(
    wrapper
      .findAll('li')
      .at(3)
      .text()
  ).toBe('Resevoir Doggies');
});
