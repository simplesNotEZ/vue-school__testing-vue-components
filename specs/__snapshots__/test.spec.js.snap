// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infinity War</strong></li>
    <li><strong>Resevoir Doggies</strong></li>
  </ul>
</div>
`;

exports[`mount the TestComponent 1`] = `
<div class="smelly">
  <h1 class="smellyFace">Hello FlueSchool</h1>
</div>
`;
